mllib/src/main/scala/org/apache/spark/ml/optim/IterativelyReweightedLeastSquares.scala

```diff
@@ -86,13 +86,11 @@ private[ml] class IterativelyReweightedLeastSquares(
         standardizeFeatures = false, standardizeLabel = false).fit(newInstances)
 
       // Check convergence
-      val oldCoefficients = oldModel.coefficients
-      val coefficients = model.coefficients
-      BLAS.axpy(-1.0, coefficients, oldCoefficients)
-      val maxTolOfCoefficients = oldCoefficients.toArray.reduce { (x, y) =>
-        math.max(math.abs(x), math.abs(y))
+      val oldCoefficients = oldModel.coefficients.toArray :+ oldModel.intercept
+      val coefficients = model.coefficients.toArray :+ model.intercept
+      val maxTol = oldCoefficients.zip(coefficients).map(x => x._1 - x._2).reduce {
```

Comment:
Is there some way to verify that this change does not impact execution time?  The BLAS.axpy operation on coefficients/oldcoefficients means that only oldcoefficients needs to be converted to an array, and the BLAS.axpy operation is probably faster than the map with subtraction, but I could be wrong.